lib/hiera/backend/http_backend.rb

```diff
@@ -38,6 +38,15 @@ def initialize
       end
 
       def lookup(key, scope, order_override, resolution_type)
+
+        # if confine_to_keys is configured, then only proceed if one of the
+        # regexes matches the lookup key
+        #
+        if @config[:confine_to_keys].is_a?(Array)
+          match = @config[:confine_to_keys].select { |r| key[Regexp.new(r)] }[0]
```

Comment:
Note: The "Ruby-ist" way to do `arr.select { ... }[0]` is `arr.find { ... }`. And if we only need a boolean value, use `arr.any? { ... }` is more readable.